data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java

```diff
@@ -105,8 +106,11 @@ protected KustoResultSetTable(JSONObject jsonTable) throws KustoServiceError {
                             String message = exceptions.getString(0);
                             throw new KustoServiceError(message);
                         } else {
-                            throw new KustoServiceError(exceptions);
+                            throw new KustoServiceError(exceptions, false, EXCEPTIONS_MESSAGE);
```

Comment:
I would change this entire block to:
`
if (exceptions != null) {
    throw new KustoServiceError(exceptions, false, EXCEPTIONS_MESSAGE);
}
`

I think it's the exception class' job to take the exceptions and set up the object accordingly, and this implementation convolutes the business logic. And of course this disregards that I don't even like the existence of the KustoServiceError class :-)
(If doing this, we'd also have to revert the EXCEPTIONS_MESSAGE to say it "failed were inner exception(s)")

Reply:
so its the same as i answered in the other one - i wanted the message of the exception to be the exception if there is only one exception